### PR TITLE
fix: improve integration override behavior and flush policy validation

### PR DESCRIPTION
**Pull request by SKannaniOS**
## Description
This PR addresses critical issues in RudderStackAnalytics SDK related to integration configuration priority and flush policy validation. The main fix ensures that user-provided integration settings correctly override default values, preventing unexpected behavior when developers want to disable specific integrations. Additionally, the PR enhances validation logic for flush policies to provide more predictable fallback behavior.

**Key Issues Fixed:**
- Fixed integration override priority in `RudderOption` where default values were incorrectly overriding user-provided settings
- Improved validation logic in `CountFlushPolicy` and `FrequencyFlushPolicy` to use more readable range checks and proper default fallbacks
- Enhanced test coverage with comprehensive test cases for edge scenarios

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Code refactor/optimization

### RudderOption Integration Priority Fix
- **Problem**: When users provided integrations like `["All": false, "Amplitude": true]`, the default integration `["All": true]` was overriding the user's `["All": false]` setting
- **Solution**: Changed the merge order from `(integrations ?? [:]) + Constants.payload.integration` to `Constants.payload.integration + (integrations ?? [:])`
- **Impact**: Now user-provided integration settings correctly override defaults, ensuring developers can properly disable integrations

### Flush Policy Validation Improvements
- **CountFlushPolicy**: Replaced `min(max())` clamping with readable range validation using `contains()` operator
- **FrequencyFlushPolicy**: Improved validation logic to use proper comparison and default fallback
- **Benefits**: More maintainable code, clearer intent, and consistent fallback behavior

### Comprehensive Test Coverage
- Added `RudderOptionTests.swift` with 8 comprehensive test cases covering:
  - Default initialization behavior
  - Custom integration override scenarios (including the critical "All" flag override)
  - Custom context and external ID handling
  - Edge cases with nil and empty parameters
- Enhanced `FlushPolicyModuleTests.swift` with additional validation tests for invalid values

## Checklist
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added the necessary documentation (if appropriate).
- [x] I have ensured that my code follows the project's code style.
- [x] I have checked for potential performance impacts and optimized if necessary.
- [x] I have checked the code for security issues.
- [ ] I have updated the changelog (if required).

### Integration Override Testing
1. Create a `RudderOption` with custom integrations that override defaults:
   ```swift
   let option = RudderOption(integrations: ["All": false, "Amplitude": true])
   ```
2. Verify that `option.integrations["All"]` returns `false` (not `true`)
3. Verify that `option.integrations["Amplitude"]` returns `true`

### Flush Policy Validation Testing
1. Test `CountFlushPolicy` with invalid values:
   ```swift
   let policy1 = CountFlushPolicy(flushAt: -5) // Should use default
   let policy2 = CountFlushPolicy(flushAt: 1000) // Should use default
   ```
2. Test `FrequencyFlushPolicy` with invalid intervals:
   ```swift
   let policy = FrequencyFlushPolicy(flushIntervalInMillis: 0) // Should use default
   ```

## Breaking Changes
This is a **non-breaking change**. The fix corrects existing behavior to match expected developer intent without changing public APIs. Existing code will continue to work, but integration override behavior will now function as originally intended.

## Maintainers Checklist
- [ ] The code has been reviewed.
- [ ] CI tests have passed.
- [ ] All necessary documentation has been updated.

## Screenshots (if applicable)
N/A - This PR contains logic fixes and test improvements with no UI changes.

## Additional Context
- **Backward Compatibility**: No breaking changes - the fix ensures the SDK behaves as developers would naturally expect
- **Test Impact**: All existing tests continue to pass, with enhanced coverage for edge cases
- **Performance**: No performance impact - only changes in validation logic and operator precedence

### CODE REVIEW

File: Sources/RudderStackAnalytics/Events/RudderOption.swift [hunk 1]
```diff
@@ -38,7 +38,7 @@ public class RudderOption: NSObject {
      The initial integrations are set to the default integrations defined in `Constants.payload.integration`.
      */
     public init(integrations: [String: Any]? = nil, customContext: [String: Any]? = nil, externalIds: [ExternalId]? = nil) {
-        self.integrations = (integrations ?? [:]) + Constants.payload.integration
+        self.integrations = Constants.payload.integration + (integrations ?? [:])
```
[the remark this answers is not in the document]
Comment (SKannaniOS) [reply Sources/RudderStackAnalytics/Events/RudderOption.swift:41]: No need. Internally `+` operator uses same `merging` call only.

**Review by 1abhishekpandey — APPROVED** [review]

**Review by ChryssaAliferi — APPROVED** [review]
